src/lsm/compaction.zig

```diff
@@ -483,18 +499,17 @@ pub fn CompactionType(
                 @src(),
             );
 
-            if (values_in[0].len == 0 and values_in[1].len == 0) {
-                compaction.input_state = .exhausted;
-            } else if (values_in[0].len == 0) {
-                compaction.copy(.b);
-            } else if (values_in[1].len == 0) {
-                if (compaction.drop_tombstones) {
-                    compaction.copy_drop_tombstones();
-                } else {
-                    compaction.copy(.a);
+            {
+                var a_iterator = ArrayIterator{ .values = compaction.values_in[0] };
+                defer compaction.values_in[0] = a_iterator.values;
+
+                var b_iterator = ArrayIterator{ .values = compaction.values_in[1] };
+                defer compaction.values_in[1] = b_iterator.values;
```

Comment:
Why is setting `values_in` necessary? Is it just to be cautious against aliasing? It doesn't look like it's used anywhere